### PR TITLE
Close #4 add section 3.18 handling of missing values

### LINKED ISSUE #4: Handling of missing values
I think a section or some notes regarding handling of missing values would be helpful.

For example, in "Creating new variables" the SAS code and the R code produce different results if `age` is missing. In SAS `AGEGR1` is set to `'<= 50 years old'` while in R it is set to `NA_character_`.

Also `!=` can give unexpected results for SAS programmers. E.g.,
```
  select(adlb, ABLFL != "Y")
```
never returns any observations because if `ABLFL` is `NA`, `ABLFL != "Y"` results in `NA` and not in `TRUE`.

### PR DESCRIPTION
**Pull request by kaz462**

### CODE REVIEW

File: 02-first_steps.Rmd [hunk 1]
```diff
@@ -240,6 +240,175 @@ adsl %>%
...
+### Handling of missing values
+
+Missing values in SAS and R:
+
+* SAS 
+  + Missing value is a blank/a single decimal point for character/numeric variables, see details in [Missing Values in SAS](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.5/lrcon/p175x77t7k6kggn1io94yedqagl3.htm);
+  
+* R
+  + Missing data in R appears as `NA`. `NA` is not a string nor a numeric value, but an indicator of missingness;
+  + `NA` and " " are different in R: " " is a blank string, while `NA` is missing;
```
Comment (bundfussr) [inline 02-first_steps.Rmd:252]: By the way, there is another difference.

In SAS trailing spaces are removed when comparing strings while this is not done in R. For example,
`" " = "'` is `1` in SAS but `" " == ""` is `FALSE` in R.

Thus I would use `""` for blank strings in R.

Comment (kaz462) [reply 02-first_steps.Rmd]: Thanks, @bundfussr ! those are valuable feedback. please let me know if further updates are needed :)

**Comment by kaz462** [comment]
new section `3.18 handling of missing values` has a summary of missing values between SAS and R + three examples of handling of missing values. please let me know if further updates are needed @bundfussr @iqis @rossfarrugia, thanks! 

- 3.1.8.1 Subsetting data: != example, as mentioned in [issue#4](https://github.com/Bayer-Group/sas2r/issues/4), [issue#12](https://github.com/Bayer-Group/sas2r/issues/12)

- 3.1.8.2 Sorting data: an example from the recent admiral testing project

- 3.1.8.3 Creating new variables: missing age example, as mentioned in [issue#4](https://github.com/Bayer-Group/sas2r/issues/4)

File: 02-first_steps.Rmd [hunk 2]
```diff
@@ -240,6 +240,177 @@ adsl %>%
...
+### Handling of missing values
+
+Missing values in SAS and R:
+
+* SAS 
+  + Missing value is a blank/a single decimal point for character/numeric variables, see details in [Missing Values in SAS](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.5/lrcon/p175x77t7k6kggn1io94yedqagl3.htm);
+  
+* R
+  + Missing data in R appears as `NA`. `NA` is not a string nor a numeric value, but an indicator of missingness;
+  + `NA` and "" are different in R: "" is a blank string, while `NA` is missing;
```
Comment (bundfussr) [inline 02-first_steps.Rmd:252]: I would use backticks for `""` to avoid typographic quotation marks.

File: 02-first_steps.Rmd [hunk 3]
```diff
@@ -240,6 +240,177 @@ adsl %>%
...
+  + "Missing values (`NA`) and `NaN` values are regarded as non-comparable even to themselves, so comparisons involving them will always result in `NA`" 
+  + `admiral::convert_blanks_to_na()` can turn SAS blank strings into proper R `NA` (from admiral documentation).
```
Comment (bundfussr) [inline 02-first_steps.Rmd:254]: I would add a link to the documentation of `convert_blanks_to_na()`: https://pharmaverse.github.io/admiral/reference/convert_blanks_to_na.html

File: 02-first_steps.Rmd [hunk 4]
```diff
@@ -240,6 +240,177 @@ adsl %>%
...
+  kable_classic_2(full_width = F)
+```
+
+
+
+
+
+#### Sorting data
+
+In SAS, missing is treated as the smallest value in `proc sort`; while in R, NA is always at the bottom after ascending or descending sorting.
```
Comment (bundfussr) [inline 02-first_steps.Rmd:323]: Backticks are missing for `NA`.

**Review by bundfussr — COMMENTED** [review]
Looks good to me.  I have added just some minor comments.

**Comment by kaz462** [comment]
thanks @bundfussr for the review! 
those issues are addressed. I also added a reference link from [Advanced R #3.2.3 Missing values](https://adv-r.hadley.nz/vectors-chap.html#missing-values)

**Review by bundfussr — APPROVED** [review]

**Pull request merged**